docs/proposals/20191030-machine-health-checking.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Machine health checking a.k.a node auto repair
+authors:
+  - "@enxebre"
+  - "@bison"
+reviewers:
+  - "@detiber"
+  - "@vincepri"
+  - "@ncdc"
+  - "@timothysc"
+creation-date: 2019-10-30
+last-updated: 2019-10-31
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Title
+- Machine health checking a.k.a node auto repair
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories [optional]](#user-stories-optional)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Examples](#examples)
+        - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+        - [Beta -> GA Graduation](#beta---ga-graduation)
+        - [Removing a deprecated flag](#removing-a-deprecated-flag)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks [optional]](#drawbacks-optional)
+  - [Alternatives [optional]](#alternatives-optional)
+  - [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+## Glossary
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+Enable opt in automated health checking and remediation of unhealthy nodes backed by machines.
+
+## Motivation
+- Reduce administrative overhead to run a cluster.
+- Ability to respond to failures of machines and keep the cluster nodes healthy.
+
+### Goals
+- Enable automated remediation for pools of machines/nodes (e.g a machineSet)
```

Comment:
```suggestion
- Enable automated remediation for groups of machines/nodes (e.g a machineSet)
```

Let's try to avoid the use of the word pools to avoid confusion with the MachinePool proposal